src/db/migrate.rs

```diff
@@ -618,6 +618,88 @@ pub fn migrate(version: Option<Version>, conn: &mut Client) -> CratesfyiResult<(
             DROP INDEX github_repos_stars_idx;
             ",
         ),
+        migration!(
+            context,
+            27,
+            // description
+            "Add gitlab handling: creation of the new repositories table which replaces and extend \
+             github_repos",
+            // upgrade query
+            "
+                CREATE TABLE repositories (
+                    id SERIAL PRIMARY KEY,
+                    host VARCHAR NOT NULL,
+                    host_id VARCHAR NOT NULL,
+                    name VARCHAR NOT NULL,
+                    description VARCHAR,
+                    last_commit TIMESTAMPTZ,
+                    stars INT NOT NULL,
+                    forks INT NOT NULL,
+                    issues INT NOT NULL,
+                    updated_at TIMESTAMPTZ NOT NULL,
+                    UNIQUE (host, host_id)
+                );
+
+                ALTER TABLE releases ADD COLUMN repository INTEGER
+                    REFERENCES repositories(id) ON DELETE SET NULL;
```

Comment:
Why `ON DELETE SET NULL`? When would we delete a repository?

Reply:
If we receive a demand to do so? I just applied the same logic than the one on `github_repos`.

Reply:
The row in `repositories` is deleted if the repository returns "not found" from the API, i.e. it has been deleted/set to private on the forge.

Reply:
Yep, @Nemo157's comment is why I added the `ON DELETE SET NULL` in my GitHub PR, otherwise we'd continue to try fetching stats for deleted repositories.